app/lib/mapLink.js

```diff
@@ -0,0 +1,36 @@
+require('object.values').shim();
+const qs = require('querystring');
+
+function joinAllTruthyValues(obj) {
+  return Object.values(obj)
+    .filter(value => value)
+    .join();
+}
+
+function addUrl(postcodeLocationDetails, inputList) {
+  return inputList.map((item) => {
+    const address = joinAllTruthyValues(item.address);
+    const saddr = `${postcodeLocationDetails.location.lat},${postcodeLocationDetails.location.lon}`;
+    let fullNameAndAddress;
+
+    if (item.name.includes('@')) {
```

Comment:
Why is this important? Maybe extract it out to a function with an informative name?